libressl_3_0: 3.0.1 -> 3.0.2

ruuda opened the pull request:
###### Motivation for this change

LibreSSL 3.0.2 has been released upstream. [Changelog](https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.0.2-relnotes.txt).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @thoughtpolice @fpletz

ruuda committed:
libressl_3_0: 3.0.1 -> 3.0.2

fpletz commented:
@GrahamcOfBorg build libressl_3_0

fpletz reviewed (approved):
I'll also backport this.

Since this is the first stable release of the 3.0 branch we could also bump the default `libressl` attribute to `libressl_3_0` on master.

ruuda committed:
libressl: 2.9.2 -> 3.0.2

ruuda committed:
libressl_2_8: remove, not maintained anymore

Stable LibreSSL releases are supported one year after their OpenBSD release.
OpenBSD 6.4 with this branch was released on 2018-10-18.

ruuda commented:
> Since this is the first stable release of the 3.0 branch we could also bump the default libressl attribute to libressl_3_0 on master.

I added a commit that changes the default to 3.0. I also added a commit to remove 2.8. [2.8 was released with OpenBSD 6.4](https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.8.2-relnotes.txt), [which was released on 2018-10-18](https://www.openbsd.org/64.html), more than one year ago. If I understand correctly, that makes 2.8 unsupported now.

fpletz commented:
Yup, exactly, thanks a lot!
